src/docker.rs

```diff
@@ -90,6 +90,29 @@ pub(crate) enum ClientType {
     SSL,
     #[cfg(windows)]
     NamedPipe,
+    Custom {
+        scheme: String,
+    },
+}
+
+type TransportReturnTy =
+    Pin<Box<dyn Future<Output = Result<Response<hyper::body::Incoming>, Error>> + Send>>;
+
+/// Callback transport trait
+pub trait TransportTrait: Send + Sync {
```

Comment:
```suggestion
/// Callback transport trait
pub trait TransportTrait: Send + Sync {
```

a nit: I personally find such (pre/post)-fixes excessive, they don't have meaningful part - the type itself defines itself as a trait.
So it's subjective opinion, but `Transport` or `CustomTransport` should work well. Also `callback` in documentation looks a bit unrelated, it's just an interface to pass your custom transport implementation. 
It's just implemented for callback too

Reply:
I'll make the changes